contributing.md

```diff
@@ -87,6 +87,16 @@ and ideally reproduce the bug. Simply encountering an error does not mean a bug
 reported; as below, search JIRA and search and inquire on the Spark user / dev mailing lists 
 first. Unreproducible bugs, or simple error reports, may be closed.
 
+It's very helpful if the bug report has a description about how the bug was introduced, by 
+which commit, so that reviewers can easily understand the bug. It also helps committers to 
+decide how far the bug fix should be backported, when the pull request is merged.
+
+Performance regression is also one kind of bug. The pull request to fix a performance regression 
+must provide a benchmark to prove the problem is indeed fixed.
```

Comment:
It should provide a benchmark of course usually. But is it a must to comply? I would rather make it `should` and leave a room for exceptions.

Reply:
It's like a bug fix must provide a regression test. Without a benchmark, how can we know the performance regression is real and it's fixed? The benchmark doesn't need to be committed to spark codebase though.

Reply:
Ah, yea. I meant the commit stuff.